crates/cubecl-wgpu/src/compute/server.rs

```diff
@@ -371,12 +381,22 @@ impl ComputeServer for WgpuServer {
         // to store this with a 'static lifetime, but the compute pass must
         // be dropped before the encoder. This isn't unsafe - it's still checked at runtime.
         let pass = self.current_pass.get_or_insert_with(|| {
-            self.encoder
+            let start_write = if !self.query_started { Some(0) } else { None };
+            let end_write = Some(1);
+            self.query_started = true;
+
+            let pass = self
+                .encoder
                 .begin_compute_pass(&wgpu::ComputePassDescriptor {
                     label: None,
-                    timestamp_writes: None,
+                    timestamp_writes: Some(wgpu::ComputePassTimestampWrites {
+                        query_set: &self.query_set,
+                        beginning_of_pass_write_index: start_write,
+                        end_of_pass_write_index: end_write,
+                    }),
```

Comment:
Do we always want to keep track of timestamps? Does it slow down compute?

Reply:
Yes :/ It's a bit unfortunate - see my comment about maybe needing a special benchark function that runs kernels in a scope, would be a bit cleaner.

That said, as far as I can measure there's no overhead, it only requires writing a 8 byte timestamp each tasks_mask kernels so not a lot of work.